lib/Sema/CSDiagnostics.h

```diff
@@ -807,20 +807,54 @@ class MissingCallFailure final : public FailureDiagnostic {
   bool diagnoseAsError() override;
 };
 
-class MissingPropertyWrapperUnwrapFailure final : public ContextualFailure {
-  DeclName PropertyName;
+class PropertyWrapperReferenceFailure : public ContextualFailure {
+  VarDecl *Property;
+  Optional<DeclName> MemberName;
```

Comment:
Actually I looked at the API for DeclName and I think it's not really necessary to wrap it into optional since it could be empty.